src/jvmMain/kotlin/io/beatmaps/api/alerts.kt

```diff
@@ -0,0 +1,138 @@
+package io.beatmaps.api
+
+import io.beatmaps.common.db.NowExpression
+import io.beatmaps.common.dbo.Alert
+import io.beatmaps.common.dbo.AlertDao
+import io.beatmaps.common.dbo.AlertRecipient
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.locations.Location
+import io.ktor.locations.get
+import io.ktor.locations.options
+import io.ktor.locations.post
+import io.ktor.request.receive
+import io.ktor.response.header
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import kotlinx.datetime.toKotlinInstant
+import org.jetbrains.exposed.sql.JoinType
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.exposed.sql.update
+
+@Location("/api/alerts") class AlertsApi {
+    @Location("/unread/{id?}")
+    data class Unread(val id: Int? = null, val api: AlertsApi)
+
+    @Location("/read/{id?}")
+    data class Read(val id: Int? = null, val api: AlertsApi)
+
+    @Location("/mark")
+    data class Mark(val api: AlertsApi)
+
+    @Location("/markall")
+    data class MarkAll(val api: AlertsApi)
+}
+
+data class MarkAlert(val id: Int, val read: Boolean)
+
+data class MarkAllAlerts(val read: Boolean)
+
+fun Route.alertsRoute() {
+    options<AlertsApi.Unread> {
```

Comment:
These options routes only need to be present on routes intended to be called from other sites (ie rabbit previewer calls /api/maps/{key} to get title / download url). At this time no authenticated routes need this.

They would also need the header adding to get requests which hasn't been done.